docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,854 @@
+
+### The Scheduler Module
+
+#### Description
+
+The Scheduler module is responsible for two main tasks:
+  - Assigning validators to parachains and parathreads.
+  - Scheduling parachains and parathreads
+
+Most of its work occurs within initialization, but it also does work within finalization.
+
+The Scheduler manages resource allocation using the concept of "Execution Cores". Each parachain has its own dedicated core, while parathreads are multiplexed over another set of cores. Parathreads also have a retrying mechanism built-in, and the main parathread cores cannot be repurposed for this use-case for economic security reasons. This gives us 3 sets of execution cores all-in-all: parachain cores, parathread cores, and parathread retry cores.
+
+An execution core is considered occupied for the entire duration it is assigned, as well as the entire time it takes for the submitted block to become available. When a core is assigned, the ownership of the core passes from the Scheduler module to the Inclusion module. If no candidate is included when the core is assigned, the core is returned to the Scheduler in block finalization. Otherwise, the core is returned when the block becomes available or the availability timeout is reached.
+
+So what should happen when an execution core is assigned and owned by the inclusion module when a change in configuration occurs? And what about assigning cores optimistically, so in the case that a core is freed by a parablock becoming available, the core can be reassigned within the span of the same block. Without optimistic assignment, the throughput of parachains is halved, as inclusion and availability can only occur in alternating blocks.
+
+
+#### Storage
+
+```rust
+ParathreadQueue: Vec<ParathreadBid>,
+ExecutionCores: Bitvec,
+```
+
+#### Initialization
+
+
+
+### The Inclusion Module
+
+### The Validity Module
+
+
+----
+
+## Processes
+
+### Processes and Jobs
+
+In this section we define the notions of Processes and Jobs. These are guidelines for how we will employ an architecture of hierarchical state machines. We'll have a top-level state machine which oversees the next level of state machines which oversee another layer of state machines and so on. The next sections will lay out these guidelines for what we've called Processes and Jobs, since this model applies to many of the tasks that the Node-side behavior needs to encompass, but these are only guidelines and some processes may have deeper hierarchies internally.
+
+Processes are long-lived worker tasks that are in charge of performing some particular kind of work. All processes can communicate with each other via a well-defined protocol. Processes can't communicate directly, but must communicate through an Overseer, which is responsible for relaying messages, handling process failures, and dispatching work signals.
```

Comment:
Do all processes have direct access to the Overseer to communicate, or do they always pass messages strictly up and down through the hierarchy?

Reply:
Processes exist one level down from the overseer in the hierarchy of asynchronous state machines. They will have a bidirectional channel with the overseer for communicating with it.